### ISSUE TITLE
Feature Request: Minimize to System Tray

### ISSUE THREAD

**Issue by Tahinli** [elevenhsoft/WebApps#28]
Hi, It would be nice, if  we can minimize apps to system tray. For example whatsapp web. It can stay open every time and get notifications.

Amount of examples can increase. Github notifications, discord notifications etc.

This will also prevent user to wait every time for loading web pages.

**Comment by elevenhsoft** [comment]
Well. I think we can't do this. Apart from COSMIC Web Apps, which (maybe in the future, because now it's also impossible) could be potentially running in tray, we can't control much behavior of web apps. Remember, they are still standalone browsers. We just saying them run behave like a "web app". For firefox, we control userChrome.css, to hide some elements, etc. 

There was an idea from someone else on Reddit to make an applet. This is possible and maybe I will do this soon. Applet for COSMIC desktop will allow you to quickly run your web apps, without opening COSMIC Web Apps. But it's future plans, and it will only be available on COSMIC de.

Current workaround for your issue is probably minimizing it to taskbar and enable all notifications.

**Comment by Tahinli** [comment]
Thanks for considering.

**Comment by elevenhsoft** [comment]
I won't close this. I keep some issues for future. Maybe someone will come with more suggestions. Let's it be here

**Comment by CarsonBurke** [comment]
I'll do some research and testing to see if I can make this happen

**Comment by lxwulf** [comment]
> Well. I think we can't do this. Apart from COSMIC Web Apps, which (maybe in the future, because now it's also impossible) could be potentially running in tray, we can't control much behavior of web apps. Remember, they are still standalone browsers. We just saying them run behave like a "web app". For firefox, we control userChrome.css, to hide some elements, etc.
> 
> There was an idea from someone else on Reddit to make an applet. This is possible and maybe I will do this soon. Applet for COSMIC desktop will allow you to quickly run your web apps, without opening COSMIC Web Apps. But it's future plans, and it will only be available on COSMIC de.
> 
> Current workaround for your issue is probably minimizing it to taskbar and enable all notifications.

Why only for COSMIC desktop environment? 

We could also make use of the tray section in the gnome desktop or not? 🤔

Would be nice to have a general solution to this instead binding to one desktop environment which also can not everywhere available directly like other environments (inside the official repos).

**Comment by hepp3n** [comment]
> Why only for COSMIC desktop environment?

Simply it's written in Rust which I'm familiar is. I wrote applet for COSMIC before, so I know how to do this I think.

> We could also make use of the tray section in the gnome desktop or not? 🤔

I don't know. GNOME does not support tray per se. You can have extensions which will display tray icons, but it's not built in solution. Also they have some sort of "tray" apps in your "control center" but it's only for flatpak applications.

> Would be nice to have a general solution to this instead binding to one desktop environment which also can not everywhere available directly like other environments (inside the official repos).

Sorry but I'm just newbie. Using Linux many years and know something about this OS, used different desktops, wms, etc.
I'm not software engineer or something. I'm doing it because I have free time and I like it.

If someone comes with something kind of general solution I would like to learn how to implement this.
My ideas was to make an applet because I know how to do this.

Also, keep in mind, those are just a desktop entries for your browser of choice. So if Firefox for example, can be set as a tray icon, your web app also would be. But unfortunately we can't do this.

libcosmic right now also does not have tray support I guess, if we have proper tray support which I think will come in future, we will be able to make Quick Web Apps as a tray app and from there you will be able to launch your apps :)

Thanks!

**Comment by lxwulf** [comment]
Many thanks @hepp3n for your detailed post! I'm looking forward to that feature confidential. We will make it work together. 💪

Its true GNOME desktop does not have a tray function directly or at least a very confusing one, but with the extension [AppIndicator and KStatusNotifierItem Support](https://github.com/ubuntu/gnome-shell-extension-appindicator) it's no problem at all if the application itself as a system tray function. Sorry, I mean no it is no difficulty to show the tray icon itself in GNOME, I do not know if the integration is easier with the extension though. 🤷‍♂️

> **Note:**
Did not mean to stress you in any way! I'm a beginner myself regarding programming. Nonetheless, I think it's not important how many experience you, or I may have. It's more significant that we have the courage and motivation to try or die while doing. And of course a goal in front of us. 😉

**Comment by hepp3n** [comment]
Yeah, when tray support will land to libcosmic or iced we can do this for sure. For now, COSMIC applet is what I can do.

I'm just happy I did something other people enjoy using 😄

**Comment by hepp3n** [comment]
Ok, so I just dig into this topic and found Iced's roadmap: https://whimsical.com/roadmap-iced-7vhq6R35Lp3TmYH4WeYwLM

Looks like it will be added in next version to Iced (tray icon). So basically, when libcosmic will rebase to this version we can have this feature working :)

**Comment by lxwulf** [comment]
Yeah, that sounds promising!
